### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by fazpu**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -92,8 +121,172 @@ export class MyComponent implements OnInit {
...
 ```
 
+#### Usage of Forms services
+- the `exampleFormService` service is a generated service holding a `FormGroup` definition that corresponds
+ with the request data structure
+
+```html
+<form [formGroup]="exampleFormService.form" (ngSubmit)="sendForm()" class="full-width">
+
+    <input type="text" name="email" required  placeholder="email"
```
Comment (jnwltr) [inline README.md]: The generated form should take care of `required`.

File: README.md [hunk 2]
```diff
@@ -92,8 +121,172 @@ export class MyComponent implements OnInit {
 }
 ```
 
+#### Usage of Forms services
+- the `exampleFormService` service is a generated service holding a `FormGroup` definition that corresponds
+ with the request data structure
+
+```html
+<form [formGroup]="exampleFormService.form" (ngSubmit)="sendForm()" class="full-width">
```
Comment (jnwltr) [inline README.md]: Shouldn't we show the component `.ts` also?

File: README.md [hunk 3]
```diff
@@ -92,8 +121,172 @@ export class MyComponent implements OnInit {
...
+
+    <input type="text" name="email" required  placeholder="email"
+           [formControl]="exampleFormService.form.get('email')" />
```
Comment (jnwltr) [inline README.md]: I would maybe demo this on `formControlName="email"`. Just consider.

File: README.md [hunk 4]
```diff
@@ -92,8 +121,172 @@ export class MyComponent implements OnInit {
...
+
+  ngOnDestroy() {
+    this.ngDestroy.next();
+    this.ngDestroy.complete();
+  }
+}
+```
+
+##### Effect (generated)
+- the dispatched method is intercepted by an effect which calls the API via the generated API service
```
Comment (jnwltr) [inline README.md]: Let's make it a sentence (capital letter, period).

File: README.md [hunk 5]
```diff
@@ -92,8 +121,172 @@ export class MyComponent implements OnInit {
...
+    this.data = this.store.select(s => ExampleState.data)
+    this.loading = this.store.select(s => ExampleState.loading)
+    this.error = this.store.select(s => ExampleState.error)
+}
+```
+
 ## Assumptions / limitations
 
+
 1. swagger file is in [version 2](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md) format, it must be json
```
Comment (jnwltr) [inline README.md]: Please add an info, that we are skipping array-like structures, so only empty `FormArray` is created in generated forms.

File: README.md [hunk 6]
```diff
@@ -104,6 +297,9 @@ generated inside Order.ts
 can contain a link to the swagger UI method reference, e.g. `http://example.com/swagger/swagger-ui.html#!/Order/Order`
 
 
+#### Usage of NGRX modules
```
Comment (jnwltr) [inline README.md]: Not sure what does this say.

File: README.md (hunk 1 above)
Comment (fazpu) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (fazpu) [reply README.md]: added

File: README.md (hunk 4 above)
Comment (fazpu) [reply README.md]: it would be inconsistent with the rest

File: README.md (hunk 6 above)
Comment (fazpu) [reply README.md]: changed to `NGRX workflow with generated modules, actions, effects, reducers and form services`

File: README.md (hunk 5 above)
Comment (fazpu) [reply README.md]: done

**Pull request merged**